server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -573,16 +573,19 @@ export class AssetService {
   private getThumbnailPath(asset: AssetEntity, format: GetAssetThumbnailFormatEnum) {
     switch (format) {
       case GetAssetThumbnailFormatEnum.WEBP:
-        if (asset.webpPath && asset.webpPath.length > 0) {
-          return asset.webpPath;
+        if (asset.webpPath) {
+          return [asset.webpPath, 'image/webp'];
         }
 
       case GetAssetThumbnailFormatEnum.JPEG:
       default:
         if (!asset.resizePath) {
-          throw new NotFoundException('resizePath not set');
+          throw new NotFoundException(`No thumbnail found for asset ${asset.id}`);
         }
-        return asset.resizePath;
+        if (format != GetAssetThumbnailFormatEnum.JPEG) {
+          this.logger.warn(`${format} thumbnail requested but not found for asset ${asset.id}, falling back to JPEG`);
```

Comment:
You can log this without the if statement if you move it up a few lines 😛

Reply:
You have a point there...